compiler/testData/diagnostics/tests/DefaultValuesTypechecking.fir.kt

```diff
@@ -8,9 +8,31 @@ fun bar(x : Int = "", y : Int = x, z : String = y) {
 // KT-371 Resolve default parameters for constructors
 
 class A(x : Int = <!UNRESOLVED_REFERENCE!>y<!>, y : Int = x) { // None of the references is resolved, no types checked
-    fun foo(bool: Boolean, a: Int = <!UNRESOLVED_REFERENCE!>b<!>, b: String = a) {}
+    fun foo(bool: Boolean, a: Int = <!UNINITIALIZED_PARAMETER!>b<!>, b: String = a) {}
 }
 
 val z = 3
 
-fun foo(x: Int = <!UNRESOLVED_REFERENCE!>y<!>, y: Int = x, i : Int = z): Int = x + y
+fun foo(x: Int = <!UNINITIALIZED_PARAMETER!>y<!>, y: Int = x, i : Int = z): Int = x + y
+
+fun foo(x: () -> Int = { <!UNINITIALIZED_PARAMETER!>y<!> }, y: Int = x(), i : Int = z): Int = x() + y
+
+fun bar(x: () -> Int = { <!UNINITIALIZED_PARAMETER!>y<!>; 1 }, y: Int) {}
+
+fun baz(
+    x: () -> Int = {
+        fun bar(xx: () -> Int = { <!UNINITIALIZED_PARAMETER!>y<!>; 1 }) = xx
+        bar()()
+    },
+    y: Int
+) {
+}
+
+fun boo(
+    x: () -> Int = {
+        fun bar(): Int = <!UNINITIALIZED_PARAMETER!>y<!>
+        bar()
+    },
+    y: Int
+) {
+}
```

Comment:
Please also add a test case with default parameter in constructors.

Reply:
I think It's already covered here: https://github.com/JetBrains/kotlin/pull/4321/files#diff-6fd8afd9d4ee9ff3a6e58a15a7c1853c4e770a919fe60faf3f8aa971907629dfR26

Reply:
But, having more wouldn't hurt, added.

Reply:
Duh, it's already there above!

https://github.com/JetBrains/kotlin/pull/4321/files#diff-785bbdf0d10bf7bf27c298b2d1bcffab38fb02e72cc1240b1b5450ff0ed3fb86R10